### PR TITLE
Update to Ember 3.8

### PR DESCRIPTION
**Pull request by andrewpye**
### What's here?
- results of running `ember-cli-update --to 3.8` and manually fixing merge conflicts;
- results of running `ember-cli-update --run-codemods`;
- removes use of deprecated `Ember.copy`;
- update to `ember-sinon-qunit` v3.4.0 and associated test syntax changes;
- fixes promise handling in `stripe` service tests so tests pass;
- switches from `npm` to `yarn`;
- replaces usages of `sweettooth` with `smile-io`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
 ---
 language: node_js
 node_js:
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
+  # so that your addon works for all apps
   - "6"
```
Comment (sivakumar-kailasam) [inline .travis.yml]:
```suggestion
  - "8"
```

File: package.json [hunk 2]
```diff
@@ -9,49 +9,58 @@
...
+    "ember-source-channel-url": "^1.1.0",
+    "ember-try": "^1.0.0",
+    "eslint-plugin-ember": "^5.2.0",
+    "eslint-plugin-node": "^7.0.1",
+    "loader.js": "^4.7.0",
+    "qunit-dom": "^0.8.0"
   },
   "engines": {
-    "node": ">= 4"
+    "node": "6.* || 8.* || >= 10.*"
```
Comment (sivakumar-kailasam) [inline package.json]:
```suggestion
    "node": "8.* || >= 10.*"
```

File: addon/components/stripe-checkout.js [hunk 3]
```diff
@@ -1,12 +1,8 @@
-import Ember from 'ember';
+import { bool } from '@ember/object/computed';
+import Component from '@ember/component';
```
Comment (vladucu) [inline addon/components/stripe-checkout.js]: nit: Component import first line

File: package.json [hunk 4]
```diff
@@ -9,49 +9,58 @@
     "stripejs",
     "sweettooth"
   ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/sweettooth/ember-cli-stripe"
```
Comment (vladucu) [inline package.json]:
```suggestion
    "url": "https://github.com/smile-io/ember-cli-stripe"
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+* `git clone <repository-url>`
+* `cd ember-cli-stripe`
+* `yarn install`
+
+## Linting
+
+* `yarn lint:hbs`
+* `yarn lint:js`
+* `yarn lint:js -- --fix`
```
Comment (vladucu) [inline CONTRIBUTING.md]:
```suggestion
* `yarn lint:js --fix`
```

File: README.md [hunk 6]
```diff
@@ -1,4 +1,7 @@
-# Stripe checkout for Ember [![Build Status](https://travis-ci.org/sweettooth/ember-cli-stripe.svg?branch=master)](http://travis-ci.org/sweettooth/ember-cli-stripe)
+ember-cli-stripe
+==============================================================================
+
+# Stripe checkout for Ember [![Build Status](https://travis-ci.org/smile-io/ember-cli-stripe.svg?branch=master)](http://travis-ci.org/smile-io/ember-cli-stripe)
```
Comment (vladucu) [inline README.md]: keeping it like this we basically have to H1's
Screen Shot 2019-07-15 at 3 33 22 PM

**Pull request merged**